src/utils/maintenance/updateAMTAdminPassword.ts

```diff
@@ -1,80 +1,109 @@
+const amt = new AMT.AMT()
+
+export const updateAMTAdminPassword = async (clientId: string, message: any, responseMsg: ClientResponseMsg, clientManager: IClientManager, configurator: IConfigurator, validator: IValidator, httpHandler: HttpHandler): Promise<ClientMsg> => {
   const logger = new Logger('updateAMTAdminPassword')
   const clientObj = clientManager.getClientObject(clientId)
-  const wsmanResponse = message
-  if (wsmanResponse?.Header?.Method) {
-    if (wsmanResponse.Body?.ReturnValue !== 0) {
-      throw new RPSError(`${wsmanResponse.Header.Method} failed for ${clientObj.uuid}`)
-    }
-    MqttProvider.publishEvent('success', ['updateAMTAdminPassword'], 'AMT admin Password updated', clientObj.uuid)
-    return true
-  } else if (wsmanResponse.AMT_GeneralSettings != null) {
-    // Response from GeneralSettings wsman call
-    const digestRealm = wsmanResponse.AMT_GeneralSettings.response.DigestRealm
-    // Validate Digest Realm
-    if (!validator.isDigestRealmValid(digestRealm)) {
-      throw new RPSError(`Device ${clientObj.uuid} activation failed. Not a valid digest realm.`)
+  switch (message.statusCode) {
+    case 401: {
+      const xmlRequestBody = amt.GeneralSettings(AMT.Methods.GET, (httpHandler.messageId++).toString())
+      const data = httpHandler.wrapIt(xmlRequestBody)
+      return responseMsg.get(clientId, data, 'wsman', 'ok', 'alls good!')
     }
-    if (!configurator?.amtDeviceRepository) {
-      MqttProvider.publishEvent('fail', ['updateAMTAdminPassword'], 'Unable to write device', clientObj.uuid)
-      logger.error('unable to write device')
-    }
-    const amtPassword: string = await configurator.profileManager.getAmtPassword(clientObj.ClientData.payload.profile.profileName)
-    clientObj.amtPassword = amtPassword
-    clientObj.ClientData.payload.digestRealm = digestRealm
-    clientObj.hostname = clientObj.ClientData.payload.hostname
-    clientManager.setClientObject(clientObj)
-    await configurator.amtDeviceRepository.insert(new AMTDeviceDTO(clientObj.uuid,
-      clientObj.hostname,
-      null,
-      null,
-      EnvReader.GlobalEnvConfig.amtusername,
-      clientObj.amtPassword,
-      null
-    ))
-    /* Register device metadata with MPS */
-    try {
-      const profile = await configurator.profileManager.getAmtProfile(clientObj.ClientData.payload.profile.profileName)
-      let tags = []
-      if (profile?.tags != null) {
-        tags = profile.tags
-      }
-      await got(`${EnvReader.GlobalEnvConfig.mpsServer}/api/v1/devices`, {
-        method: 'POST',
-        json: {
-          guid: clientObj.uuid,
-          hostname: clientObj.hostname,
-          mpsusername: clientObj.mpsUsername,
-          tags: tags,
-          tenantId: profile.tenantId
+    case 200: {
+      const xmlBody = parseBody(message)
```

Comment:
Probably a good idea around parsing to have a try catch

Reply:
done